src/pages/ReportParticipantsPage.tsx

```diff
@@ -1,75 +1,82 @@
+import useLocalize from '@hooks/useLocalize';
 import useThemeStyles from '@hooks/useThemeStyles';
 import compose from '@libs/compose';
 import * as LocalePhoneNumber from '@libs/LocalePhoneNumber';
+import type * as Localize from '@libs/Localize';
 import Navigation from '@libs/Navigation/Navigation';
 import * as PersonalDetailsUtils from '@libs/PersonalDetailsUtils';
 import * as ReportUtils from '@libs/ReportUtils';
+import type {OptionData} from '@libs/ReportUtils';
 import * as UserUtils from '@libs/UserUtils';
 import CONST from '@src/CONST';
+import type {TranslationPaths} from '@src/languages/types';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
+import type {PersonalDetails, Report} from '@src/types/onyx';
+import type { ValueOf } from 'type-fest';
+import type {WithReportOrNotFoundProps} from './home/report/withReportOrNotFound';
 import withReportOrNotFound from './home/report/withReportOrNotFound';
-import personalDetailsPropType from './personalDetailsPropType';
-import reportPropTypes from './reportPropTypes';
 
-const propTypes = {
-    /* Onyx Props */
-
-    /** The personal details of the person who is logged in */
-    personalDetails: PropTypes.objectOf(personalDetailsPropType),
-
-    /** The active report */
-    report: reportPropTypes.isRequired,
-
-    /** Route params */
-    route: PropTypes.shape({
-        params: PropTypes.shape({
-            /** Report ID passed via route r/:reportID/participants */
-            reportID: PropTypes.string,
-        }),
-    }).isRequired,
-
-    ...withLocalizePropTypes,
+type ReportParticipantsPageOnyxProps = {
+    personalDetails: OnyxCollection<PersonalDetails>;
 };
 
-const defaultProps = {
-    personalDetails: {},
-};
+type ReportParticipantsPageProps = ReportParticipantsPageOnyxProps & WithReportOrNotFoundProps;
+
+type ParticipantData = {
+    alternateText: string;
+    displayName: string;
+    accountID: number;
+    icons: Array<{
+        id: number;
+        source: UserUtils.AvatarSource;
+        name: string;
+        type: ValueOf<typeof CONST.ICON_TYPE_AVATAR>;
+    }>;
+    keyForList: string;
+    login: string;
+    text: string;
+    tooltipText: string;
+    participantsList: Array<{
+        accountID: number;
+        displayName: string;
+    }>;            
+}
 
 /**
  * Returns all the participants in the active report
  *
- * @param {Object} report The active report object
- * @param {Object} personalDetails The personal details of the users
- * @param {Object} translate The localize
- * @return {Array}
+ * @param report The active report object
+ * @param personalDetails The personal details of the users
+ * @param translate The localize
  */
-const getAllParticipants = (report, personalDetails, translate) =>
-    _.chain(ReportUtils.getVisibleMemberIDs(report))
+const getAllParticipants = (
+    report: OnyxEntry<Report>,
+    personalDetails: OnyxCollection<PersonalDetails>,
```

Comment:
You can use `PersonalDetailsList` type

Reply:
@JKobrynski `personalDetails` is not an Onyx collection, so we should use `OnyxEntry<PersonalDetailsList>`

Reply:
Ok, got it! Thanks guys